File: src/apis/__generated/model/enums/DictConstants.ts

export const DictConstants = {
    COUPON_TYPE: 1006,
    PAY_TYPE: 1004,
    MENU_TYPE: 1002,
    REFUND_STATUS: 1010,
    PRODUCT_ORDER_STATUS: 1003,
    GENDER: 1001,
    COUPON_SCOPE_TYPE: 1007,
    COUPON_USE_STATUS: 1008,
    COUPON_RECEIVE_TYPE: 1009,
    NAVIGATOR_TYPE: 1013,
    QUALITY_TYPE: 1014,
    ORDER_TYPE: 1012,
    USER_STATUS: 1005,
}
export const Dictionaries = {
    CouponType: {
        DISCOUNT: {
          keyId: 0,
          keyName: '折扣券',
          keyEnName: 'DISCOUNT',
          dictId: 1006,
          dictName: '优惠券类型',
          dictEnName: 'COUPON_TYPE',
          orderNum: 0
        },
        REDUCE: {
          keyId: 1,
          keyName: '满减券',
          keyEnName: 'REDUCE',
          dictId: 1006,
          dictName: '优惠券类型',
          dictEnName: 'COUPON_TYPE',
          orderNum: 0
        },
      },
    PayType: {
        WE_CHAT_PAY: {
          keyId: 0,
          keyName: '微信支付',
          keyEnName: 'WE_CHAT_PAY',
          dictId: 1004,
          dictName: '支付类型',
          dictEnName: 'PAY_TYPE',
          orderNum: 0
        },
      },
    MenuType: {
        BUTTON: {
          keyId: 2,
          keyName: '按钮',
          keyEnName: 'BUTTON',
          dictId: 1002,
          dictName: '菜单类型',
          dictEnName: 'MENU_TYPE',
          orderNum: 2
        },
        PAGE: {
          keyId: 0,
          keyName: '页面',
          keyEnName: 'PAGE',
          dictId: 1002,
          dictName: '菜单类型',
          dictEnName: 'MENU_TYPE',
          orderNum: 1
        },
        DIRECTORY: {
          keyId: 1,
          keyName: '目录',
          keyEnName: 'DIRECTORY',
          dictId: 1002,
          dictName: '菜单类型',
          dictEnName: 'MENU_TYPE',
          orderNum: 0
        },
      },
    RefundStatus: {
        REFUNDING: {
          keyId: 0,
          keyName: '退款中',
          keyEnName: 'REFUNDING',
          dictId: 1010,
          dictName: '退款状态',
          dictEnName: 'REFUND_STATUS',
          orderNum: 0
        },
        SUCCESS: {
          keyId: 1,
          keyName: '退款成功',
          keyEnName: 'SUCCESS',
          dictId: 1010,
          dictName: '退款状态',
          dictEnName: 'REFUND_STATUS',
          orderNum: 0
        },
        FAILED: {
          keyId: 2,
          keyName: '退款失败',
          keyEnName: 'FAILED',
          dictId: 1010,
          dictName: '退款状态',
          dictEnName: 'REFUND_STATUS',
          orderNum: 0
        },
      },
    ProductOrderStatus: {
        CLOSED: {
          keyId: 4,
          keyName: '已关闭',
          keyEnName: 'CLOSED',
          dictId: 1003,
          dictName: '商品订单状态',
          dictEnName: 'PRODUCT_ORDER_STATUS',
          orderNum: 0
        },
        REFUNDED: {
          keyId: 5,
          keyName: '已退款',
          keyEnName: 'REFUNDED',
          dictId: 1003,
          dictName: '商品订单状态',
          dictEnName: 'PRODUCT_ORDER_STATUS',
          orderNum: 0
        },
        TO_BE_RECEIVED: {
          keyId: 2,
          keyName: '待收货',
          keyEnName: 'TO_BE_RECEIVED',
          dictId: 1003,
          dictName: '商品订单状态',
          dictEnName: 'PRODUCT_ORDER_STATUS',
          orderNum: 0
        },
        TO_BE_PAID: {
          keyId: 0,
          keyName: '待付款',
          keyEnName: 'TO_BE_PAID',
          dictId: 1003,
          dictName: '商品订单状态',
          dictEnName: 'PRODUCT_ORDER_STATUS',
          orderNum: 0
        },
        TO_BE_EVALUATED: {
          keyId: 3,
          keyName: '待评价',
          keyEnName: 'TO_BE_EVALUATED',
          dictId: 1003,
          dictName: '商品订单状态',
          dictEnName: 'PRODUCT_ORDER_STATUS',
          orderNum: 0
        },
        FINISHED: {
          keyId: 6,
          keyName: '已完成',
          keyEnName: 'FINISHED',
          dictId: 1003,
          dictName: '商品订单状态',
          dictEnName: 'PRODUCT_ORDER_STATUS',
          orderNum: 0
        },
        TO_BE_DELIVERED: {
          keyId: 1,
          keyName: '待发货',
          keyEnName: 'TO_BE_DELIVERED',
          dictId: 1003,
          dictName: '商品订单状态	',
          dictEnName: 'PRODUCT_ORDER_STATUS',
          orderNum: 0
        },
      },
    Gender: {
        PRIVATE: {
          keyId: 2,
          keyName: '保密',
          keyEnName: 'PRIVATE',
          dictId: 1001,
          dictName: '性别',
          dictEnName: 'GENDER',
          orderNum: 0
        },
        MALE: {
          keyId: 0,
          keyName: '男',
          keyEnName: 'MALE',
          dictId: 1001,
          dictName: '性别',
          dictEnName: 'GENDER',
          orderNum: 0
        },
        FEMALE: {
          keyId: 1,
          keyName: '女',
          keyEnName: 'FEMALE',
          dictId: 1001,
          dictName: '性别',
          dictEnName: 'GENDER',
          orderNum: 1
        },
      },
    CouponScopeType: {
        CATEGORY: {
          keyId: 2,
          keyName: '品类券',
          keyEnName: 'CATEGORY',
          dictId: 1007,
          dictName: '优惠券使用范围',
          dictEnName: 'COUPON_SCOPE_TYPE',
          orderNum: 0
        },
        PRODUCT: {
          keyId: 1,
          keyName: '商品券',
          keyEnName: 'PRODUCT',
          dictId: 1007,
          dictName: '优惠券使用范围',
          dictEnName: 'COUPON_SCOPE_TYPE',
          orderNum: 0
        },
        GENERAL: {
          keyId: 0,
          keyName: '通用券',
          keyEnName: 'GENERAL',
          dictId: 1007,
          dictName: '优惠券使用范围',
          dictEnName: 'COUPON_SCOPE_TYPE',
          orderNum: 0
        },
      },
    CouponUseStatus: {
        USED: {
          keyId: 1,
          keyName: '已使用',
          keyEnName: 'USED',
          dictId: 1008,
          dictName: '优惠券使用状态',
          dictEnName: 'COUPON_USE_STATUS',
          orderNum: 0
        },
        EXPIRED: {
          keyId: 2,
          keyName: '已过期',
          keyEnName: 'EXPIRED',
          dictId: 1008,
          dictName: '优惠券使用状态',
          dictEnName: 'COUPON_USE_STATUS',
          orderNum: 0
        },
        UNUSED: {
          keyId: 0,
          keyName: '未使用',
          keyEnName: 'UNUSED',
          dictId: 1008,
          dictName: '优惠券使用状态',
          dictEnName: 'COUPON_USE_STATUS',
          orderNum: 0
        },
      },
    CouponReceiveType: {
        GIFT: {
          keyId: 0,
          keyName: '系统赠送',
          keyEnName: 'GIFT',
          dictId: 1009,
          dictName: '优惠券获取方式',
          dictEnName: 'COUPON_RECEIVE_TYPE',
          orderNum: 0
        },
      },
    NavigatorType: {
        PRODUCT: {
          keyId: 0,
          keyName: '商品',
          keyEnName: 'PRODUCT',
          dictId: 1013,
          dictName: '跳转类型',
          dictEnName: 'NAVIGATOR_TYPE',
          orderNum: 0
        },
        BLIND_BOX: {
          keyId: 1,
          keyName: '盲盒',
          keyEnName: 'BLIND_BOX',
          dictId: 1013,
          dictName: '跳转类型',
          dictEnName: 'NAVIGATOR_TYPE',
          orderNum: 0
        },
      },
    QualityType: {
        GENERAL: {
          keyId: 0,
          keyName: '普通款',
          keyEnName: 'GENERAL',
          dictId: 1014,
          dictName: '商品品质',
          dictEnName: 'QUALITY_TYPE',
          orderNum: 0
        },
        HIDDEN: {
          keyId: 1,
          keyName: '隐藏款',
          keyEnName: 'HIDDEN',
          dictId: 1014,
          dictName: '商品品质',
          dictEnName: 'QUALITY_TYPE',
          orderNum: 0
        },
        LEGENDARY: {
          keyId: 2,
          keyName: '超神款',
          keyEnName: 'LEGENDARY',
          dictId: 1014,
          dictName: '商品品质',
          dictEnName: 'QUALITY_TYPE',
          orderNum: 0
        },
      },
    OrderType: {
        BLIND_BOX_ORDER: {
          keyId: 1,
          keyName: '盲盒订单',
          keyEnName: 'BLIND_BOX_ORDER',
          dictId: 1012,
          dictName: '订单类型',
          dictEnName: 'ORDER_TYPE',
          orderNum: 0
        },
        PRODUCT_ORDER: {
          keyId: 0,
          keyName: '商品订单',
          keyEnName: 'PRODUCT_ORDER',
          dictId: 1012,
          dictName: '订单类型',
          dictEnName: 'ORDER_TYPE',
          orderNum: 0
        },
        VIP_ORDER: {
          keyId: 2,
          keyName: 'VIP订单',
          keyEnName: 'VIP_ORDER',
          dictId: 1012,
          dictName: '订单类型',
          dictEnName: 'ORDER_TYPE',
          orderNum: 0
        },
      },
    UserStatus: {
        NORMAL: {
          keyId: 0,
          keyName: '正常',
          keyEnName: 'NORMAL',
          dictId: 1005,
          dictName: '用户状态',
          dictEnName: 'USER_STATUS',
          orderNum: 0
        },
        BANNED: {
          keyId: 1,
          keyName: '封禁',
          keyEnName: 'BANNED',
          dictId: 1005,
          dictName: '用户状态',
          dictEnName: 'USER_STATUS',
          orderNum: 0
        },
      },
}

File: server/src/main/java/kaigee/top/dict/model/DictConstants.java
package kaigee.top.dict.model;
import lombok.Getter;
import lombok.AllArgsConstructor;

public class DictConstants {
  public static final String PAY_TYPE = "PAY_TYPE";
  public static final String MENU_TYPE = "MENU_TYPE";
  public static final String REFUND_STATUS = "REFUND_STATUS";
  public static final String PRODUCT_ORDER_STATUS = "PRODUCT_ORDER_STATUS";
  public static final String COUPON_SCOPE_TYPE = "COUPON_SCOPE_TYPE";
  public static final String COUPON_USE_STATUS = "COUPON_USE_STATUS";
  public static final String COUPON_RECEIVE_TYPE = "COUPON_RECEIVE_TYPE";
  public static final String SEX = "SEX";
  public static final String NAVIGATOR_TYPE = "NAVIGATOR_TYPE";
  public static final String QUALITY_TYPE = "QUALITY_TYPE";
  public static final String COUPON_TYPE = "COUPON_TYPE";
  public static final String ORDER_TYPE = "ORDER_TYPE";
  public static final String USER_STATUS = "USER_STATUS";
  @Getter
  @AllArgsConstructor
  public enum PayType{
        WE_CHAT_PAY(0, "微信支付", "WE_CHAT_PAY", 1004, "支付类型", "PAY_TYPE", 0),
  ;
  final int keyId;
  final String keyName;
  final String keyEnName;
  final int dictId;
  final String dictName;
  final String dictEnName;
  final int orderNum;
  }
  @Getter
  @AllArgsConstructor
  public enum MenuType{
        BUTTON(2, "按钮", "BUTTON", 1002, "菜单类型", "MENU_TYPE", 2),
        PAGE(0, "页面", "PAGE", 1002, "菜单类型", "MENU_TYPE", 1),
        DIRECTORY(1, "目录", "DIRECTORY", 1002, "菜单类型", "MENU_TYPE", 0),
  ;
  final int keyId;
  final String keyName;
  final String keyEnName;
  final int dictId;
  final String dictName;
  final String dictEnName;
  final int orderNum;
  }
  @Getter
  @AllArgsConstructor
  public enum RefundStatus{
        REFUNDING(0, "退款中", "REFUNDING", 1010, "退款状态", "REFUND_STATUS", 0),
        SUCCESS(1, "退款成功", "SUCCESS", 1010, "退款状态", "REFUND_STATUS", 0),
        FAILED(2, "退款失败", "FAILED", 1010, "退款状态", "REFUND_STATUS", 0),
  ;
  final int keyId;
  final String keyName;
  final String keyEnName;
  final int dictId;
  final String dictName;
  final String dictEnName;
  final int orderNum;
  }
  @Getter
  @AllArgsConstructor
  public enum ProductOrderStatus{
        CLOSED(4, "已关闭", "CLOSED", 1003, "商品订单状态", "PRODUCT_ORDER_STATUS", 0),
        REFUNDED(5, "已退款", "REFUNDED", 1003, "商品订单状态", "PRODUCT_ORDER_STATUS", 0),
        TO_BE_RECEIVED(2, "待收货", "TO_BE_RECEIVED", 1003, "商品订单状态", "PRODUCT_ORDER_STATUS", 0),
        TO_BE_PAID(0, "待付款", "TO_BE_PAID", 1003, "商品订单状态", "PRODUCT_ORDER_STATUS", 0),
        TO_BE_EVALUATED(3, "待评价", "TO_BE_EVALUATED", 1003, "商品订单状态", "PRODUCT_ORDER_STATUS", 0),
        FINISHED(6, "已完成", "FINISHED", 1003, "商品订单状态", "PRODUCT_ORDER_STATUS", 0),
        TO_BE_DELIVERED(1, "待发货", "TO_BE_DELIVERED", 1003, "商品订单状态	", "PRODUCT_ORDER_STATUS", 0),
  ;
  final int keyId;
  final String keyName;
  final String keyEnName;
  final int dictId;
  final String dictName;
  final String dictEnName;
  final int orderNum;
  }
  @Getter
  @AllArgsConstructor
  public enum CouponScopeType{
        CATEGORY(2, "品类券", "CATEGORY", 1007, "优惠券使用范围", "COUPON_SCOPE_TYPE", 0),
        PRODUCT(1, "商品券", "PRODUCT", 1007, "优惠券使用范围", "COUPON_SCOPE_TYPE", 0),
        GENERAL(0, "通用券", "GENERAL", 1007, "优惠券使用范围", "COUPON_SCOPE_TYPE", 0),
  ;
  final int keyId;
  final String keyName;
  final String keyEnName;
  final int dictId;
  final String dictName;
  final String dictEnName;
  final int orderNum;
  }
  @Getter
  @AllArgsConstructor
  public enum CouponUseStatus{
        USED(1, "已使用", "USED", 1008, "优惠券使用状态", "COUPON_USE_STATUS", 0),
        EXPIRED(2, "已过期", "EXPIRED", 1008, "优惠券使用状态", "COUPON_USE_STATUS", 0),
        UNUSED(0, "未使用", "UNUSED", 1008, "优惠券使用状态", "COUPON_USE_STATUS", 0),
  ;
  final int keyId;
  final String keyName;
  final String keyEnName;
  final int dictId;
  final String dictName;
  final String dictEnName;
  final int orderNum;
  }
  @Getter
  @AllArgsConstructor
  public enum CouponReceiveType{
        GIFT(0, "系统赠送", "GIFT", 1009, "优惠券获取方式", "COUPON_RECEIVE_TYPE", 0),
  ;
  final int keyId;
  final String keyName;
  final String keyEnName;
  final int dictId;
  final String dictName;
  final String dictEnName;
  final int orderNum;
  }
  @Getter
  @AllArgsConstructor
  public enum SEX{
        FEMALE(1, "女", "FEMALE", 1001, "性别", "SEX", 0),
        MALE(0, "男", "MALE", 1001, "性别", "SEX", 0),
  ;
  final int keyId;
  final String keyName;
  final String keyEnName;
  final int dictId;
  final String dictName;
  final String dictEnName;
  final int orderNum;
  }
  @Getter
  @AllArgsConstructor
  public enum NavigatorType{
        PRODUCT(0, "商品", "PRODUCT", 1013, "跳转类型", "NAVIGATOR_TYPE", 0),
        BLIND_BOX(1, "盲盒", "BLIND_BOX", 1013, "跳转类型", "NAVIGATOR_TYPE", 0),
  ;
  final int keyId;
  final String keyName;
  final String keyEnName;
  final int dictId;
  final String dictName;
  final String dictEnName;
  final int orderNum;
  }
  @Getter
  @AllArgsConstructor
  public enum QualityType{
        GENERAL(0, "普通款", "GENERAL", 1014, "商品品质", "QUALITY_TYPE", 0),
        HIDDEN(1, "隐藏款", "HIDDEN", 1014, "商品品质", "QUALITY_TYPE", 0),
        LEGENDARY(2, "超神款", "LEGENDARY", 1014, "商品品质", "QUALITY_TYPE", 0),
  ;
  final int keyId;
  final String keyName;
  final String keyEnName;
  final int dictId;
  final String dictName;
  final String dictEnName;
  final int orderNum;
  }
  @Getter
  @AllArgsConstructor
  public enum CouponType{
        REDUCE(1, "满减券", "REDUCE", 1006, "优惠券类型", "COUPON_TYPE", 0),
  ;
  final int keyId;
  final String keyName;
  final String keyEnName;
  final int dictId;
  final String dictName;
  final String dictEnName;
  final int orderNum;
  }
  @Getter
  @AllArgsConstructor
  public enum OrderType{
        BLIND_BOX_ORDER(1, "盲盒订单", "BLIND_BOX_ORDER", 1012, "订单类型", "ORDER_TYPE", 0),
        PRODUCT_ORDER(0, "商品订单", "PRODUCT_ORDER", 1012, "订单类型", "ORDER_TYPE", 0),
        VIP_ORDER(2, "VIP订单", "VIP_ORDER", 1012, "订单类型", "ORDER_TYPE", 0),
  ;
  final int keyId;
  final String keyName;
  final String keyEnName;
  final int dictId;
  final String dictName;
  final String dictEnName;
  final int orderNum;
  }
  @Getter
  @AllArgsConstructor
  public enum UserStatus{
        NORMAL(0, "正常", "NORMAL", 1005, "用户状态", "USER_STATUS", 0),
        BANNED(1, "封禁", "BANNED", 1005, "用户状态", "USER_STATUS", 0),
  ;
  final int keyId;
  final String keyName;
  final String keyEnName;
  final int dictId;
  final String dictName;
  final String dictEnName;
  final int orderNum;
  }
}
